rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -230,48 +229,51 @@ def _features_for_patterns(
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional[Metadata] = None,
-        cached_component: Optional["RegexFeaturizer"] = None,
-        should_finetune: bool = False,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "RegexFeaturizer":
-        """Loads a previously trained component.
-
-        Args:
-            meta: Configuration of trained component.
-            model_dir: Path where trained pipeline is stored.
-            model_metadata: Metadata for the trained pipeline.
-            cached_component: Previously cached component(if any).
-            should_finetune: Indicates whether to load the component for further
-                finetuning.
-            **kwargs: Any other arguments.
-        """
-        file_name = meta["file"]
-
-        patterns_file_name = Path(model_dir) / (file_name + ".patterns.pkl")
-
+    ) -> RegexFeaturizerGraphComponent:
+        """Loads trained component (see parent class for full docstring)."""
         known_patterns = None
-        if patterns_file_name.exists():
-            known_patterns = rasa.shared.utils.io.read_json_file(patterns_file_name)
 
-        return RegexFeaturizer(
-            meta, known_patterns=known_patterns, finetune_mode=should_finetune,
-        )
+        try:
+            with model_storage.read_from(resource) as model_dir:
+                patterns_file_name = model_dir / "patterns.pkl"
+                if patterns_file_name.exists():
```

Comment:
Don't we want to warn if that resource isn't there?   ( Sometimes I've added a "this might be because it wasn't trained" - because I had kept the "do not persist if not trained" approach as it was before)

Reply:
There wasn't one before that's why haven't added one. Will add a debug message

Reply:
the case actually can't happen. So handling it as `FileNotFoundError` now